server/src/main/java/com/unosquare/carmigo/service/PlatformUserService.java

```diff
@@ -70,7 +72,7 @@ public void deletePlatformUserById(final int id)
         platformUserRepository.deleteById(id);
     }
 
-    public GrabDriverDTO getDriverById(int id)
+    public GrabDriverDTO getDriverById(final int id)
```

Comment:
You can delete from repository directly and then handle the exception if the driver does not exist.  Otherwise you have two calls to the DB here.

Reply:
I don't understand.

Reply:
You have two lines of code:


```
Driver driver = driverRepository.findById(id)
                .orElseThrow(() -> new ResourceNotFoundException("Driver id " + id + " not found."));

driverRepository.deleteById(id);
```

Each one is making a call to the DB.  You only need to do the delete.  If it can not be found then hibernate will throw an exception which you can catch with a generic not found.